### PR TITLE
fix: Tooltip type improvements

### PR DESCRIPTION
**Pull request by dannygoldstein**
Follow-on to #1002. 

* Uses correct (non-vega escaped) column names to access `rowType.propertyTypes` object, fixing crashes
* Adds fallback (non-caching) behavior for missing keys
* Refactors tooltip cache path

### CODE REVIEW

**Review by gtarpenning — APPROVED** [review]
Fantastic, looks resolved: 
Screenshot 2023-12-22 at 4 05 59 PM

File: weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx [hunk 1]
```diff
@@ -110,6 +111,33 @@ const PLOT_TEMPLATE: VisualizationSpec = {
   },
 };
 
+function tooltipNoCache(
+  row: any,
+  s: SeriesConfig,
+  flatResultNode: Node,
+  weave: Weave
```
Comment (jamie-rasmussen) [inline weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx:118]: Could the last parameter just be `opStore: OpStore` - that seems like all you need.

File: weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx [hunk 2]
```diff
@@ -758,10 +786,10 @@ export const PanelPlot2Inner: React.FC<PanelPlotProps> = props => {
       concattedNonLineTable,
       concreteConfig.series,
       vegaReadyTables,
+      listOfTableNodes,
       vegaCols,
       flatResultNode,
-      weave.client.opStore,
-      listOfTableNodes,
+      weave,
```
Comment (jamie-rasmussen) [inline weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx:792]: How about `const {opStore} = weave.client` at the top of the function and then just use `opStore` everywhere?

**Review by jamie-rasmussen — APPROVED** [review]

File: weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx (hunk 2 above)
Comment (dannygoldstein) [reply weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx]: done

File: weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx (hunk 1 above)
Comment (dannygoldstein) [reply weave-js/src/components/Panel2/PanelPlot/PanelPlot2Inner.tsx]: done

**Pull request merged**